File: result/translate/chatgpt/java2python/python/FIND_THREE_ELEMENT_FROM_DIFFERENT_THREE_ARRAYS_SUCH_THAT_THAT_A_B_C_K_1.py
def f_gold(a1, a2, a3, n1, n2, n3, sum):
    s = set()
    
    for i in range(n1):
        s.add(a1[i])
    
    al = list(s)
    
    for i in range(n2):
        for j in range(n3):
            if (sum - a2[i] - a3[j]) in al and al.index(sum - a2[i] - a3[j]) != al[-1]:
                return True
                
    return False

File: result/translate/starchat/java2cpp/c++/FIND_THREE_ELEMENT_FROM_DIFFERENT_THREE_ARRAYS_SUCH_THAT_THAT_A_B_C_K_1.cpp
bool f_gold (int a1[], int a2[], int a3[], int n1, int n2, int n3, int sum) {
        set<int> s;
        for (int i = 0; i < n1; i++) s.insert(a1[i]);
        vector<int> al(s.begin(), s.end());
        for (int i = 0; i < n2; i++) {
            for (int j = 0; j < n3; j++) {
                if (find(al.begin(), al.end(), sum - a2[i] - a3[j])!= al.end() && al.back()!= sum - a2[i] - a3[j]) return true;
            }
        }
        return false;
    }